ArcGISRuntimeSDKQt_SampleViewers/ArcGISRuntimeSDKQt_Samples/qml/CategoryGridView.qml

```diff
@@ -88,4 +94,15 @@ Page {
     // Ensure virtual keyboard is not persisted after this item has been
     // hidden. This is an issue on iOS and Android devices.
     onVisibleChanged:  searchBar.focus = false;
+
+    Timer {
+        id: searchQueryEventTimer
+        interval: 2500 // milliseconds
+        running: false
+        repeat: false
+        onTriggered: {
+            if (searchBar.text !== "" && GAnalytics.telemetryEnabled)
```

Comment:
I don't know if this makes any real difference, but a bool check should be faster than a string check. I would write the bool check first so it can short circuit and avoid any string check at all if telemetry is disabled. You need to check both anyways.

Actually, another idea is to allow the API to be used with impunity but the calls all become no-ops if telemetry is disabled. That moves the responsibility of managing state exclusively in the C++ implementation. This could be a big refactor and I don't know if there is scope for that.

If you do that, code like this can simply always call `GAnalytics.postEvent`, but that logic determines if anything should happen or not.

Reply:
I have it set up so `postEvent` only runs if `telemetryEnabled == false`, so it's handled on the C++ side rather than needing to remember to check in QML.